### PR TITLE
[TEST] sam read benchmark.

### PR DESCRIPTION
**Pull request by smehringer**
supersedes #1023

### CODE REVIEW

File: test/performance/io/format_sam_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+#if SEQAN3_HAS_SEQAN2
+// ============================================================================
+// seqan2 read from stream
+// ============================================================================
+
+void seqan2_sam_file_read_from_stream(benchmark::State &state)
+{
+    size_t n_queries = state.range(0);
+    seqan3::test::tmp_filename file_name{"tmp.sam"};
```
Comment (wvandertoorn) [inline test/performance/io/format_sam_benchmark.cpp:150]: I am not too familiar with the seqan2/seqan3 mixed support, is it okay to have a seqan3 functionality in a seqan2 function?

File: test/performance/io/format_sam_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        }
+
+        clear(header);
+    }
+}
+
+void seqan2_sam_file_read_from_disk(benchmark::State &state)
+{
+    size_t n_queries = state.range(0);
+    seqan3::test::tmp_filename file_name{"tmp.sam"};
```
Comment (wvandertoorn) [inline test/performance/io/format_sam_benchmark.cpp:187]: same here

File: test/performance/io/format_sam_benchmark.cpp (hunk 1 above)
Comment (smehringer) [reply test/performance/io/format_sam_benchmark.cpp:150]: Yes it is. You can use seqan3 and seqan side by side (if they don't cause trouble). And in this case it's not part of the performance measurement, so I am also not cheating :p

File: test/performance/io/format_sam_benchmark.cpp [hunk 3]
```diff
@@ -0,0 +1,225 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
```
Comment (eseiler) [inline test/performance/io/format_sam_benchmark.cpp:6]:
```suggestion
// -----------------------------------------------------------------------------------------------------
// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
// -----------------------------------------------------------------------------------------------------
```

File: test/performance/io/format_sam_benchmark.cpp [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+// !\author Clemens Cords (clemens.cords@fu-berlin.de)
```
Comment (eseiler) [inline test/performance/io/format_sam_benchmark.cpp:8]:
```suggestion
//!\author Clemens Cords (clemens.cords@fu-berlin.de)
```

The only non header file with author information 🤔

File: test/performance/io/format_sam_benchmark.cpp [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alignment/pairwise/alignment_result.hpp>
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/alphabet/nucleotide/dna15.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/io/alignment_file/output.hpp>
+#include <seqan3/io/alignment_file/input.hpp>
+#include <seqan3/io/alignment_file/format_sam.hpp>
```
Comment (eseiler) [inline test/performance/io/format_sam_benchmark.cpp:22]:
```suggestion
#include <seqan3/io/alignment_file/input.hpp>
#include <seqan3/io/alignment_file/format_sam.hpp>
#include <seqan3/io/alignment_file/output.hpp>
```

File: test/performance/io/format_sam_benchmark.cpp [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+#include <seqan/bam_io.h>
+#endif
+
+// ============================================================================
+// generate sam file from randomly generated sequence and store in file_dict
+// ============================================================================
+
+std::map<std::size_t, std::string> file_dict;
+
+static std::string create_sam_file_string(std::size_t n_queries)
```
Comment (eseiler) [inline test/performance/io/format_sam_benchmark.cpp:37]:
```suggestion
static std::string create_sam_file_string(size_t const n_queries)
```

File: test/performance/io/format_sam_benchmark.cpp [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+                                 seqan3::sam_flag::none);                // field::flag
+        }
+
+        file_dict[n_queries] = stream.str();
+    }
+
+    return file_dict[n_queries];
+}
+
+void write_file(std::string file_name, size_t n_queries)
```
Comment (eseiler) [inline test/performance/io/format_sam_benchmark.cpp:91]:
```suggestion
void write_file(std::string const & file_name, size_t const n_queries)
```

File: test/performance/io/format_sam_benchmark.cpp [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    }
+}
+
+void sam_file_read_from_disk(benchmark::State &state)
+{
+    size_t n_queries = state.range(0);
+    seqan3::test::tmp_filename file_name{"tmp.sam"};
+    auto tmp_path = file_name.get_path();
+
+    write_file(file_name.get_path(), n_queries);
```
Comment (eseiler) [inline test/performance/io/format_sam_benchmark.cpp:129]:
```suggestion
    write_file(tmp_path, n_queries);
```

File: test/performance/io/format_sam_benchmark.cpp (hunk 4 above)
Comment (smehringer) [reply test/performance/io/format_sam_benchmark.cpp:8]: I know.. but I felt bad removing it..

**Pull request merged**